src/test/java/com/elmsoftware/env/EnvironmentSettingsModuleTest.java

```diff
@@ -118,17 +108,55 @@ public void should_use_settings_provider_if_provided() throws Exception {
 		assertNotNull(providerField);
 		providerField.setAccessible(true);
 		assertNotNull(providerField.get(module));
+	}
 
+	@Test
+	public void should_use_environment_settings_if_set() throws Exception {
+		// setup test
+		final String dockerEnvironmentVariable = "I'm configurable for docker!";
+		final EnvironmentOverridingPostProcessor postProcessor = new EnvironmentOverridingPostProcessor();
+		final EnvironmentSettingsModule module = new EnvironmentSettingsModule(new NoOpSettingProvider(), util, postProcessor);
+
+		System.setProperty("environment.json", "guice-test.json");
+		when(binder.bind(any(Key.class))).thenReturn(builder);
+		when(binder.skipSources(Names.class)).thenReturn(binder);
+
+		// run test
+		withEnvironmentVariable("SOME_KEY", dockerEnvironmentVariable)
+			.execute(() -> module.configure(binder));
+
+		// verify outcome
+		verify(builder).toInstance(eq(dockerEnvironmentVariable));
 	}
 
-	private Key<String> findKey(final List<Key<String>> allValues, final String name) {
-		for (final Key<String> key : allValues) {
-			final Named annotation = (Named) key.getAnnotation();
-			if (annotation.value().equals(name)) {
-				return key;
+	@Test
+	public void should_use_legacy_configure_on_only_string_properties() {
```

Comment:
this is a weird test, and i'm not proud of it. it's goal is mostly just to make sure property values that aren't strings don't get passed in to the legacy `configure(Binder, Map<String, String>)` implementation.